pkg/curatedpackages/kubeclient.go

```diff
@@ -0,0 +1,57 @@
+package curatedpackages
+
+import (
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	packagesv1 "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+)
+
+// NewKubeClientFromFilename creates a controller-runtime k8s client for use
+// by CLI commands.
+//
+// If the RESTConfigurator is nil, a default configurator will be used.
+func NewKubeClientFromFilename(kubeConfigFilename string, rc RESTConfigurator) (client.Client, error) {
+	data, err := os.ReadFile(kubeConfigFilename)
+	if err != nil {
+		return nil, err
+	}
+	if rc == nil {
+		rc = restConfigurator(clientcmd.RESTConfigFromKubeConfig)
+	}
+	restConfig, err := rc.Config(data)
+	if err != nil {
+		return nil, err
+	}
+	scheme := runtime.NewScheme()
+	utilruntime.Must(packagesv1.AddToScheme(scheme))
```

Comment:
nit: I wonder if we should just return an error instead of panic here
Although I could buy the argument this should only fail in case of a programing error

Reply:
I have no strong opinion, I was just copying what I've seen (specifically other places in the codebase manager/main.go). But I suppose since I can easily return an error, I should. I'll tweak.